src/link/Elf.zig

```diff
@@ -1005,14 +1003,12 @@ pub fn markDirty(self: *Elf, shdr_index: u16, phdr_index: ?u16) void {
 
 pub fn flush(self: *Elf, comp: *Compilation, prog_node: *std.Progress.Node) link.File.FlushError!void {
     if (self.base.options.emit == null) {
-        if (build_options.have_llvm) {
-            if (self.llvm_object) |llvm_object| {
-                return try llvm_object.flushModule(comp, prog_node);
-            }
+        if (self.llvm_object) |llvm_object| {
+            return try llvm_object.flushModule(comp, prog_node);
         }
         return;
     }
-    const use_lld = build_options.have_llvm and self.base.options.use_lld;
```

Comment:
I think this was yet another bug and we really ought to check for both `options.use_llvm and options.use_lld`. Otherwise, we permit combination of no-LLVM backend with LLD which makes no sense.

Reply:
Following the offline discussion on Discord, I think we should change it back to `build_options.have_llvm and options.use_lld` since as @jacobly0 rightly pointed out LLD is not available without having LLVM.

Reply:
This will apply to Coff and Elf, but not to MachO where we don't use LLD at all for example.

Reply:
I kept `have_llvm` and added `use_llvm`.